### PR TITLE
Add sub actions

### PR DESCRIPTION
**Pull request by simonsso**

### CODE REVIEW

**Review by trsmarc — CHANGES_REQUESTED** [review]
The overall looks great and I understand the general direction, but I have a couple of questions about the file structure. Would you mind explaining the reasoning behind having three Dockerfiles and two entrypoint.sh files? I'm wondering if there might be a way to consolidate them for maintainability.

**Comment by simonsso** [comment]
> The overall looks great and I understand the general direction, but I have a couple of questions about the file structure. Would you mind explaining the reasoning behind having three Dockerfiles and two entrypoint.sh files? I'm wondering if there might be a way to consolidate them for maintainability.

I know there are copies, this is just quick and fast implementation to solve this. Feel free to fix it but there is no practical benefit of a clean up from a maintenance point of view. 

One just edit all files with vim and run:
```
:argdo %s/v0.5.1/v0.6.1/|update
```

**Comment by ETeissonniere** [comment]
@simonsso @trsmarc we can have one Dockerfile and entrypoint set to `try-runtime`, then each action can define its own `args` and pass them to the image.

Doing so will allow us to reuse the current prebuild step which will greatly improve CI time, and also reduce the number of docker images to pull when running CI - which also avoids CI taking too long.

File: entrypoint.sh [hunk 1]
```diff
@@ -12,4 +12,6 @@ if [ ! -f "$snap" ]; then
   try-runtime --runtime existing create-snapshot --uri $url $snap
 fi
 
-try-runtime --runtime $runtime on-runtime-upgrade --checks=$checks $* snap --path $snap
+if [ -f "$runtime" ];then
+  try-runtime --runtime $runtime on-runtime-upgrade --checks=$checks $* snap --path $snap
+fi
```
Comment (ETeissonniere) [inline entrypoint.sh:17]: I am not sure this will do what you want since if `runtime == SKIP` this will still evaluate to true?

File: action.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ inputs:
     description: 'Path to a previously taken snapshot, if does not exists we will create it'
     default: '/tmp/default.snap'
   runtime:
-    description: 'Path to a try-runtime enabled runtime to test migrations for'
+    description: 'Path to a try-runtime enabled runtime to test migrations for, Set to SKIP to skip test and only download snapshot'
     required: true
```
Comment (ETeissonniere) [inline action.yml:15]: You can change this to `false` if you want to make this option optional

File: entrypoint.sh (hunk 1 above)
Comment (simonsso) [reply entrypoint.sh:17]: check `man [`

then scroll down to;

```
       -f FILE
              FILE exists and is a regular file
```

Any non existing argument will skip

File: action.yml (hunk 2 above)
Comment (simonsso) [reply action.yml:15]: There is a complication in how it is transferred to the Dockers' entrypoint with `$1 $2 $3 $4`

**Comment by simonsso** [comment]
> @simonsso @trsmarc we can have one Dockerfile and entrypoint set to `try-runtime`, then each action can define its own `args` and pass them to the image.
> 
> Doing so will allow us to reuse the current prebuild step which will greatly improve CI time, and also reduce the number of docker images to pull when running CI - which also avoids CI taking too long.

Maybe, I would just get his to work before we spend time to optimize it. I intend to put the actions in different CI jobs and foresee other bottle necks to have larger impact first.

The docker container is recreated in a few seconds in a job which will run for 10-20 minutes once every day or so. Not worth optimizing.

**Comment by simonsso** [comment]
> The overall looks great and I understand the general direction, but I have a couple of questions about the file structure. Would you mind explaining the reasoning behind having three Dockerfiles and two entrypoint.sh files? I'm wondering if there might be a way to consolidate them for maintainability.

I created a ticket to look into this on a rainy day
CHA-454

**Pull request closed without merging**